fedimint-server/src/net/peers.rs

```diff
@@ -383,6 +442,13 @@ where
                     },
                 }
             },
+            Some(status_query) = self.status_query_receiver.recv() => {
+                if status_query.response.send(PeerStatus::Connected).is_err() {
+                    let peer_id = self.peer;
+                    warn!(target: LOG_NET_PEER, %peer_id, "Could not send peer status response: receiver dropped");
```

Comment:
We generally have this system of signaling global shutdown after which threads start voluntarily finishing, and corresponding senders might not be able to send any more messages

In such circumstances they should either ignore (maybe log `debug!`) not being able to send, or even better exit themselves (not always possible).